cmd/gitserver/server/internal/accesslog/accesslog_test.go

```diff
@@ -0,0 +1,132 @@
+package accesslog
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/sourcegraph/log/logtest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
+	"github.com/sourcegraph/sourcegraph/internal/requestclient"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+func TestRecord(t *testing.T) {
+	t.Run("OK", func(t *testing.T) {
+		ctx := context.Background()
+		ctx = withContext(ctx, &paramsContext{})
+		Record(ctx, "github.com/foo/bar", []string{"git", "grep", "foo"})
+		pc := fromContext(ctx)
+		assert.NotNil(t, pc)
```

Comment:
nit:
```suggestion
		require.NotNil(t, pc)
```

Otherwise the test would panic on a failed assertion, but not a much big deal though.